packages/suite/src/components/suite/CoinjoinSessionCountdownTimer.tsx

```diff
@@ -1,23 +1,35 @@
-import React from 'react';
-import { useCountdownTimer, useSelector } from '@suite-hooks';
-import { getSessionDeadlineFormat } from '@wallet-utils/coinjoinUtils';
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import { useSelector } from '@suite-hooks';
 import { selectRoundsDurationInHours } from '@wallet-reducers/coinjoinReducer';
-import hoursToMilliseconds from 'date-fns/hoursToMilliseconds';
+import differenceInHours from 'date-fns/differenceInHours';
+import { FormattedNumber } from 'react-intl';
 
 interface CoinjoinSessionCountdownTimerProps {
     sessionDeadline: number;
+    unitDisplay?: 'long' | 'short' | 'narrow';
 }
 
 export const CoinjoinSessionCountdownTimer = ({
     sessionDeadline,
+    unitDisplay = 'narrow',
 }: CoinjoinSessionCountdownTimerProps) => {
+    const deadlineDate = useMemo(() => new Date(sessionDeadline), [sessionDeadline]);
     const roundsDurationInHours = useSelector(selectRoundsDurationInHours);
-
-    const { timeLeftString } = useCountdownTimer(
-        sessionDeadline,
-        getSessionDeadlineFormat(sessionDeadline),
```

Comment:
Why not picking the format based on the either `sessionDeadline` or `Date.now() + roundsDurationInHours`? Same can be done with the deadline? Then this logic would be here and not in the timeout timer, which should not care about session-specific logic but you could keep the `useCountdownTimer` hook and not rewrite the whole `CoinjoinSessionCountdownTimer`